android/java/org/chromium/chrome/browser/crypto_wallet/model/SolanaInstructionPresenter.java

```diff
@@ -0,0 +1,127 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.model;
+
+import org.chromium.brave_wallet.mojom.CoinType;
+import org.chromium.brave_wallet.mojom.SolanaAccountMeta;
+import org.chromium.brave_wallet.mojom.SolanaInstruction;
+import org.chromium.brave_wallet.mojom.SolanaInstructionAccountParam;
+import org.chromium.brave_wallet.mojom.SolanaInstructionParam;
+import org.chromium.chrome.browser.crypto_wallet.adapters.TwoLineItemRecyclerViewAdapter.TwoLineItemText;
+import org.chromium.chrome.browser.crypto_wallet.util.TransactionUtils;
+import org.chromium.chrome.browser.crypto_wallet.util.Utils;
+import org.chromium.chrome.browser.crypto_wallet.util.WalletConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class SolanaInstructionPresenter {
+    private boolean isDecodedDataPresent;
+    public List<SolanaInstructionAccountPresenter> mAccountDatas;
+    private SolanaInstruction mSolanaInstruction;
+    public boolean mIsUnknown;
+
+    public SolanaInstructionPresenter(SolanaInstruction solanaInstruction) {
+        assert solanaInstruction != null : "solanaInstruction is null";
+        mIsUnknown = TransactionUtils.isSolTxUnknown(solanaInstruction.programId);
+        mSolanaInstruction = solanaInstruction;
+        mAccountDatas = new ArrayList<>();
+        if (solanaInstruction.accountMetas != null) {
+            if (!mIsUnknown) {
```

Comment:
could be moved under one `if`

Reply:
updated: https://github.com/brave/brave-core/pull/15389/commits/0c925e12fee38080c6ba602895d813a2d98e489d